edge/stack/stack.go

```diff
@@ -521,6 +610,10 @@ func buildEnvVarsForDeployer(envVars []portainer.Pair) []string {
 }
 
 func (manager *StackManager) deleteStack(ctx context.Context, stack *edgeStack, stackName, stackFileLocation string) {
+	manager.mu.Lock()
```

Comment:
Why is this lock needed here?

Reply:
becayse we're changing the stack status. I thought it's needed

Reply:
It was needed for this line that is not there anymore
```go
delete(manager.stacks, edgeStackID(stack.ID))
```

I'm not seeing any mutation to the `manager` state here, but I could be missing it.

Reply:
`stack *edgeStack` is an object from the manager state, so changing it means changing the manager state. isn't it?